faiss/IndexIVF.cpp

```diff
@@ -352,6 +352,8 @@ void IndexIVF::search_preassigned (idx_t n, const float *x, idx_t k,
                                    IndexIVFStats *ivf_stats) const
 {
     long nprobe = params ? params->nprobe : this->nprobe;
+    nprobe = std::min((long)nlist, nprobe);
```

Comment:
looks good. While you are at it, could you replace occurrences of `long` with `idx_t` in this function ? We are trying to get rid of long, as its size is not portable.